### PR TITLE
feat: Pool ctor to lead to valid object through .load

### PR DESCRIPTION
**Pull request by Arrowana**

### CODE REVIEW

File: ts-client/src/pool.ts [hunk 1]
```diff
@@ -104,35 +107,48 @@ class Pool {
...
     ]);
 
     const parsedClockAccount = (parsedClock.value!.data as ParsedAccountData)
       .parsed as ParsedClockState;
 
-    this.vaultASpl.fromAccountsInfo(vaultATokenVault!, vaultALpMint!);
-    this.vaultBSpl.fromAccountsInfo(vaultBTokenVault!, vaultBLpMint!);
-    this.poolSpl.fromAccountsInfo(aVaultLp!, bVaultLp!, lpMint!);
-    this.onChainTime = parsedClockAccount.info.unixTimestamp;
+    // What is the meaning of those default objects?
```
Comment (codewithgun) [inline ts-client/src/pool.ts:125]: It is a class which hold the token/mint account of pool / vault.

File: ts-client/src/pool.ts [hunk 2]
```diff
@@ -39,59 +46,55 @@ function fromTokenMintsToCompositeKey(
...
+    private poolSpl: PoolSpl,
+    private onChainTime: number = 0
+  ) {
+    if ("stable" in this.state.curveType) {
+      this.swapCurve = new StableSwap(
+        this.state.curveType.stable.amp.toNumber()
+      );
+    } else {
+      this.swapCurve = new ConstantProductSwap();
+    }
```
Comment (00xSam) [inline ts-client/src/pool.ts:68]: Can we convert this to getter?

```
get swapCurve() {
      invariant(this.state.curveType, 'Curve Type not found')
      return "stable" in this.state.curveType ? new StableSwap(this.state.curveType.stable.amp.toNumber()) : new ConstantProductSwap();
}

Comment (Arrowana) [reply ts-client/src/pool.ts:68]: It is better to throw in the constructor if the created object isn't valid. Is there any advantage in making this fail later?

Comment (00xSam) [reply ts-client/src/pool.ts:68]: ohh no, just though of to save an extra declaration. I think your point make more sense

**Review by 00xSam — APPROVED** [review]
LGTM

**Pull request merged**